connectors-e2e-test/connector-http-json-test/src/test/java/io/camunda/connector/e2e/rest/HttpJsonConnectorTest.java

```diff
@@ -0,0 +1,226 @@
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.e2e.rest;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.matching;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static io.camunda.connector.test.e2e.BpmnFile.Replace.replace;
+import static io.camunda.connector.test.e2e.BpmnFile.replace;
+import static io.camunda.zeebe.process.test.assertions.BpmnAssert.assertThat;
+
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import io.camunda.connector.http.base.auth.BasicAuthentication;
+import io.camunda.connector.http.base.auth.BearerAuthentication;
+import io.camunda.connector.http.base.auth.OAuthAuthentication;
+import io.camunda.connector.http.base.constants.Constants;
+import io.camunda.connector.test.e2e.BpmnFile;
+import io.camunda.connector.test.e2e.ElementTemplate;
+import io.camunda.connector.test.e2e.ZeebeTest;
+import io.camunda.connector.test.e2e.app.TestConnectorRuntimeApplication;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.spring.test.ZeebeSpringTest;
+import java.io.File;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(
+    classes = {TestConnectorRuntimeApplication.class},
+    properties = {
+      "spring.main.allow-bean-definition-overriding=true",
+      "camunda.connector.webhook.enabled=false",
+      "camunda.connector.polling.enabled=false"
+    })
+@ZeebeSpringTest
+@ExtendWith(MockitoExtension.class)
+// TODO discuss whether we want to enable it by default
+// @EnabledIfEnvironmentVariable(named = "CONNECTORS_E2E_TEST", matches = "true")
```

Comment:
Yes, that would be great 👍  We could then add a new pipeline for running e2e tests or add a checkbox to optionally run it during snapshot / release builds.